cmd/ipfs/daemon.go

```diff
@@ -153,9 +153,18 @@ func daemonFunc(req cmds.Request, res cmds.Response) {
 		fmt.Printf("IPNS mounted at: %s\n", nsdir)
 	}
 
+	var rootRedirect corehttp.ServeOption
+	if len(cfg.Gateway.RootRedirect) > 0 {
+		rootRedirect = corehttp.RedirectOption("", cfg.Gateway.RootRedirect)
+	}
+
 	if gatewayMaddr != nil {
 		go func() {
-			err := corehttp.ListenAndServe(node, gatewayMaddr.String(), corehttp.GatewayOption)
+			var opts = []corehttp.ServeOption{corehttp.GatewayOption}
+			if rootRedirect != nil {
+				opts = append(opts, rootRedirect)
+			}
+			err := corehttp.ListenAndServe(node, gatewayMaddr.String(), opts...)
```

Comment:
this may belong in another PR, but this is the part we need to be able to start and stop on command. let's sync up with @briantigerchow on how best to do that